base_layer/wallet/src/ffi.rs

```diff
@@ -29,87 +29,402 @@
+    if bytes.is_null() {
+        let b = Box::from_raw(bytes);
+    }
+}
+
+/// adds c_char to internal vec
+pub unsafe extern "C" fn bytevector_add_byte(bytes: *mut ByteVector, byte: c_uchar) {
+    (*bytes).0.push(byte);
+}
+
+/// returns c_char at position in internal vector
+#[no_mangle]
+pub unsafe extern "C" fn bytevector_get_at(ptr: *mut ByteVector, i: c_int) -> c_uchar {
+    (*ptr).0.clone()[i as usize]
+}
+
+/// Returns the number of items, zero-indexed
+#[no_mangle]
+pub unsafe extern "C" fn bytevector_get_length(vec: *const ByteVector) -> c_int {
+    if vec.is_null() {
+        return 0;
+    }
+    (&*vec).0.len() as c_int
+}
+
+/// -------------------------------------------------------------------------------------------- ///
+
+/// -------------------------------- Public Key ------------------------------------------------ ///
+pub type WalletPrivateKey = PublicKey;
+
+#[no_mangle]
+pub unsafe extern "C" fn public_key_create(bytes: *mut ByteVector) -> *mut WalletPublicKey {
+    let mut v = Vec::new();
+    if !bytes.is_null() {
+        v = (*bytes).0.clone();
+    }
+    let pk = WalletPublicKey::from_bytes(&v).unwrap();
+    Box::into_raw(Box::new(pk))
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn public_key_destroy(pk: *mut WalletPublicKey) {
+    if !pk.is_null() {
+        Box::from_raw(pk);
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn public_key_get_key(pk: *mut WalletPublicKey) -> *mut ByteVector {
+    let mut bytes = ByteVector(Vec::new());
+    if !pk.is_null() {
+        bytes.0 = (*pk).to_vec();
+    }
+    Box::into_raw(Box::new(bytes))
+}
+
+/// -------------------------------------------------------------------------------------------- ///
+
+/// -------------------------------- Private Key ----------------------------------------------- ///
+pub type WalletPublicKey = PrivateKey;
+
+#[no_mangle]
+pub unsafe extern "C" fn privatekey_create(bytes: *mut ByteVector) -> *mut WalletPrivateKey {
+    let mut v = Vec::new();
+    if !bytes.is_null() {
+        v = (*bytes).0.clone();
+    }
+    let pk = WalletPrivateKey::from_bytes(&v).unwrap();
+    Box::into_raw(Box::new(pk))
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn privatekey_destroy(pk: *mut WalletPrivateKey) {
+    if !pk.is_null() {
+        Box::from_raw(pk);
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn privatekey_get_key(pk: *mut WalletPrivateKey) -> *mut ByteVector {
+    let mut bytes = ByteVector(Vec::new());
+    if !pk.is_null() {
+        bytes.0 = (*pk).to_vec();
+    }
+    Box::into_raw(Box::new(bytes))
+}
+
+/// -------------------------------------------------------------------------------------------- ///
+
+/// -------------------------------------- OutputManagerConfig --------------------------------- ///
+pub type WalletOutputManagerConfig = OutputManagerConfig;
+
+#[no_mangle]
+pub unsafe extern "C" fn outputmanagerconfig_create(
+    key: *mut PrivateKey,
+    b_seed: *mut c_char,
```

Comment:
Rather make the argument names descriptive i.e. `master_key` and `branch_seed`. This contraction might save you a few keystrokes but anyone who has to work with this later will have to spend time digging into the code to figure out what these argument means where a more descriptive variable name is self documenting. This also breaks the link between these arguments and the ones used in other places in the codebase.

Reply:
Fixed in the next commit, arguments are more descriptive